### PR TITLE
TBout analyzer

### PR DESCRIPTION
**Pull request by tschuh**
This PR adds an EDAnalyzer to study the output of ProducerTBout.

### CODE REVIEW

**Comment by tomalin** [comment]
This PR contains lots of code changes that were already includes in Thomas's previous PRs. (e.g. The line "channelAssignment_ = nullptr;" in L1FPGATrackProducer.cc). So its review must wait until they have been merged and a rebase has been done.

**Comment by tomalin** [comment]
Please rebase this (using the instructions for the "complicated" case from https://twiki.cern.ch/twiki/bin/view/CMS/L1TrackSoftware), since code from merged earlier PRs must be excluded.

**Comment by tschuh** [comment]
Just merged it with L1TK-dev-12_0_0_pre4

File: L1Trigger/TrackFindingTracklet/test/AnalyzerTBout.cc
Comment (tomalin) [inline L1Trigger/TrackFindingTracklet/test/AnalyzerTBout.cc]: This comment explaining what the class sounds mysleading "analyzer HW". Isn't it just checking the performance of the TBoutProducer?

Comment (tschuh) [reply L1Trigger/TrackFindingTracklet/test/AnalyzerTBout.cc]: yes and TBoutProducer delivers a hardware like structured TTStub Collection

**Review by tomalin — APPROVED** [review]
Looks good

**Pull request merged**
